src/wallet/scriptpubkeyman.cpp

```diff
@@ -1694,11 +1694,8 @@ util::Result<CTxDestination> DescriptorScriptPubKeyMan::GetNewDestination(const
         }
 
         CTxDestination dest;
-        std::optional<OutputType> out_script_type = m_wallet_descriptor.descriptor->GetOutputType();
-        if (out_script_type && out_script_type == type) {
-            ExtractDestination(scripts_temp[0], dest);
-        } else {
-            throw std::runtime_error(std::string(__func__) + ": Types are inconsistent. Stored type does not match type of newly generated address");
+        if (!ExtractDestination(scripts_temp[0], dest)) {
+            return util::Error{_("Error: Cannot extract destination from the generated scriptpubkey")}; // shouldn't happen
```

Comment:
nit: To be consistent with other internal errors that should never happen (e.g. "types are inconsistent" above), could also throw `std::runtime_error` here:
```suggestion
            throw std::runtime_error(std::string(__func__) + "Cannot extract destination from the generated scriptpubkey"); // shouldn't happen
```

Reply:
Following that logic, we should either
- get rid of literally _all_ runtime error throws that could happen in the GUI (alone in ./src/wallet there are 75 instances of those, according to `$ git grep "throw std::runtime_error" ./src/wallet | wc -l`), or
- simply catch runtime errors in the GUI, with a clear error message with dialog to the user and a clean shutdown to avoid db corruption

The second option seems to be the more obvious solution to me, rather than avoiding `std::runtime_error` (which seems a perfect candidate for internal errors that should never happen) at all.
(Of course, this is potential material for follow-ups and not directly related to this PR)